actions/java-gradle-build-push-jib-multi-plaftorm/action.yaml

```diff
@@ -69,28 +58,48 @@ runs:
         lfs: ${{ inputs.download-lfs-files }}
 
     - name: Set up Gradle with version ${{ inputs.gradle-version }}
-      uses: bakdata/ci-templates/actions/java-gradle-setup@1.42.0
+      uses: bakdata/ci-templates/actions/java-gradle-setup@tiedemann/adjust-with-metadata-action
       with:
         java-distribution: ${{ inputs.java-distribution }}
         java-version: ${{ inputs.java-version }}
         gradle-version: ${{ inputs.gradle-version }}
         gradle-cache: ${{ inputs.gradle-cache }}
         gradle-cache-read-only: ${{ inputs.gradle-cache-read-only }}
 
+    - name: Process inputs
+      id: process-inputs
+      run: |
+        # Extract the image name from the full image name
+        echo "IMAGE_NAME=$(echo "${{ inputs.full-image-name }}" | cut -d ':' -f 1)" >> "$GITHUB_OUTPUT"
+        IFS=',' read -r -a tags <<< "${{ inputs.additional-tags }}"
+        echo "IMAGE_TAGS<<EOF" >> "$GITHUB_OUTPUT"
+        for element in "${tags[@]}"
+        do
+            echo "type=raw,$element" >> $GITHUB_OUTPUT
+        done
+        echo "EOF" >> "$GITHUB_OUTPUT"
+      shell: bash
+
+    - name: Docker metadata
+      id: meta
+      uses: docker/metadata-action@v5
+      with:
+        images: ${{ steps.process-inputs.outputs.IMAGE_NAME }}
+        tags: |
+          type=ref,event=branch
+          type=sha
+          ${{ steps.process-inputs.outputs.IMAGE_TAGS }}
+
     - name: Build Docker image
       run: |
-        ./gradlew ${{ inputs.subproject && format('{0}:', inputs.subproject) || '' }}jib \
+        ./gradlew ${{ inputs.subproject && format('{0}:', inputs.subproject) || '' }} jib \
           --info --stacktrace \
-          --image=${{ inputs.image-name }} \
-          ${{ inputs.jib-from-image && format('-Djib.from.image={0}', inputs.jib-from-image) || '' }} \
+          --image=${{github.repository}}_${{ github.run_id }}_${{github.run_number}} \
           -Djib.from.platforms=linux/arm64,linux/amd64 \
-          -DsendCredentialsOverHttp \
-          -Djib.allowInsecureRegistries=true \
-          -Djib.to.image=${{ inputs.full-image-name }}:${{ inputs.image-tag }} \
-          -Djib.to.tags=${{ inputs.image-tag }} \
-          -Djib.to.auth.username=AWS \
-          -Djib.to.auth.password=${{ inputs.registry-password }} \
-          -x test \
+          ${{ inputs.jib-from-image && format('-Djib.from.image={0}', inputs.jib-from-image) || '' }} \
+          ${{ inputs.class && format('-Djib.container.mainClass={0}', inputs.class) || '' }} \
+          -Djib.to.image=${{ inputs.full-image-name }} \
```

Comment:
Please add `-Djib.allowInsecureRegistries=true` and overwrite the original [java-gradle-build-jib](https://github.com/bakdata/ci-templates/tree/main/actions/java-gradle-build-jib) action as this is what Philipp requested. Otherwise action is tested and it works. Only needs the insecure registries flag to be added and `packages: write` to be added under permissions in the workflow that calls the action, so jib can use the docker config